fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -937,5 +939,86 @@ void testRecordUpdateReducedMergeForcingAgileTimeQuota() {
         assertTrue(newLength < oldLength);
         assertTrue(timeCommitCount > 0);
     }
+
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    @ParameterizedTest
+    @BooleanSource
+    void luceneOnlineIndexingByIndexTest(boolean reverseScan) {
+        int numRecords = 47;
+        int transactionLimit = 10;
+        int groupingCount = 1;
+        final Random rn = new Random();
+        rn.nextInt();
+        Index srcIndex = new Index("src_index", field("doc_id"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS);
+        Index tgtIndex = new Index(
+                "Map_with_auto_complete$entry-value",
+                new GroupingKeyExpression(field("entry",
+                        KeyExpression.FanType.FanOut).nest(concat(LuceneIndexTest.keys)), groupingCount + 1),
+                LuceneIndexTypes.LUCENE,
+                ImmutableMap.of());
+
+        long[] docIds = new long[numRecords * 2];
+        for (int i = 0; i < numRecords * 2; i ++) {
+            docIds[i] = rn.nextLong();
+        }
+        int middle = 1 + rn.nextInt(numRecords - 2);
+        // write records
+        RecordMetaDataHook hook = metaDataBuilder -> {
+            metaDataBuilder.addIndex(MAP_DOC, srcIndex);
+            metaDataBuilder.addIndex(MAP_DOC, tgtIndex);
+            metaDataBuilder.removeIndex(TextIndexTestUtils.SIMPLE_DEFAULT_NAME);
+            TextIndexTestUtils.addRecordTypePrefix(metaDataBuilder);
+        };
+        try (final FDBRecordContext context = openContext()) {
+            recordStore = LuceneIndexTestUtils.openRecordStore(context, path, hook);
+            for (int i = 0; i < numRecords; i ++) {
+                long docId = docIds[i];
+                recordStore.saveRecord(createSimpleDocument(docId, randomText(rn), randomGroup(rn)));
+            }
+            context.commit();
+        }
+        // overwrite some records (to enforce merge), write others as new
+        try (final FDBRecordContext context = openContext()) {
+            recordStore = LuceneIndexTestUtils.openRecordStore(context, path, hook);
+            for (int i = 0; i < middle; i ++) {
+                long docId = docIds[i];
+                recordStore.saveRecord(createSimpleDocument(docId, randomText(rn), randomGroup(rn)));
+            }
+            for (int i = middle; i < numRecords; i ++) {
+                long docId = docIds[numRecords + i];
+                recordStore.saveRecord(createSimpleDocument(docId, randomText(rn), randomGroup(rn)));
+            }
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            recordStore = LuceneIndexTestUtils.openRecordStore(context, path, hook);
+            recordStore.markIndexDisabled(tgtIndex).join();
+            context.commit();
+        }
+        // build the index ..
+        try (final FDBRecordContext context = openContext()) {
```

Comment:
A good followup once #2495 is in would be to enable partitioning on the index, and assert that there are many partitions, but if we scanned in reverse, there was no partition rebalancing.